### PR TITLE
Adding folder management plugin for terraform provider

### PR DESCRIPTION
**Pull request by mpandya-sumo-logic**
Adding folder management plugin for terraform provider. Tested using acceptance tests and folder CRUD on production deployment.

Used these references for CRUD operations:
Create: https://nite-api.sumologic.net/docs/#operation/createFolder
Read: https://nite-api.sumologic.net/docs/#operation/getFolder
Update: https://nite-api.sumologic.net/docs/#operation/updateFolder
Delete: https://nite-api.sumologic.net/docs/#operation/beginAsyncDelete, https://nite-api.sumologic.net/docs/#operation/getAsyncDeleteStatus

As folder management is V2 (not V1), I updated the URLs in sumologic_client.go and request paths extensions in the respective files.

@frankreno

### CODE REVIEW

File: sumologic/provider.go [hunk 1]
```diff
@@ -47,6 +38,7 @@ func Provider() terraform.ResourceProvider {
 			"sumologic_user":                               resourceSumologicUser(),
 			"sumologic_ingest_budget":                      resourceSumologicIngestBudget(),
 			"sumologic_collector_ingest_budget_assignment": resourceSumologicCollectorIngestBudgetAssignment(),
+			"sumologic_folder": resourceSumologicFolder(),
```
Comment (frankreno) [inline sumologic/provider.go]: nit: formatting, can we indent this over to align

File: sumologic/sumologic_folder.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	response, _, err := s.Get(fmt.Sprintf("v2/content/%s/delete/%s/status", id, job_id))
+	err = json.Unmarshal(response, &deletionStatus)
+	if err != nil {
+		return "", err
+	}
+	return deletionStatus.Status, nil
+}
+
+type Folder struct {
+	Name              	string                    `json:"name"`
```
Comment (frankreno) [inline sumologic/sumologic_folder.go]: nit: alignment

File: sumologic/provider.go (hunk 1 above)
Comment (mpandya-sumo-logic) [reply sumologic/provider.go]: Updated

File: sumologic/sumologic_folder.go (hunk 2 above)
Comment (mpandya-sumo-logic) [reply sumologic/sumologic_folder.go]: Updated

**Pull request merged**